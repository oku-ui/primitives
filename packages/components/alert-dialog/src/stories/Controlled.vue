<script setup lang="ts">
import {
  OkuAlertDialog,
  OkuAlertDialogAction,
  OkuAlertDialogCancel,
  OkuAlertDialogContent,
  OkuAlertDialogDescription,
  OkuAlertDialogOverlay,
  OkuAlertDialogPortal,
  OkuAlertDialogTitle,
  OkuAlertDialogTrigger,
} from '@oku-ui/alert-dialog'
import { ref } from 'vue'

const open = ref(false)
const housePurchased = ref(false)
</script>

<template>
  <div>
    <div>
      <img
        src="https://images.unsplash.com/photo-1682686580224-cd46ea1a6950?ixlib=rb-4.0.3&ixid=M3wxMjA3fDF8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D&auto=format&fit=crop&w=500&q=80"
        alt="a large white house with a red roof"
      >
    </div>
    <OkuAlertDialog :open="open" @open-change="open = $event">
      <OkuAlertDialogTrigger
        class="alert-dialog-trigger"
        @click="(e) => {
          console.log('housePurchased', housePurchased)
          if (housePurchased) {
            e.preventDefault();
            housePurchased = false;
          }
        }"
      >
        {{ housePurchased ? 'You bought the house! Sell it!' : 'Buy this house' }}
      </OkuAlertDialogTrigger>
      <OkuAlertDialogPortal>
        <OkuAlertDialogOverlay class="alert-dialog-overlay" />
        <OkuAlertDialogContent class="alert-dialog-content">
          <OkuAlertDialogTitle class="alert-dialog-title">
            Are you sure?
          </OkuAlertDialogTitle>
          <OkuAlertDialogDescription class="alert-dialog-description">
            This will do a very dangerous thing. Thar be dragons!
          </OkuAlertDialogDescription>
          <OkuAlertDialogAction class="alert-dialog-action" @click="housePurchased = true">
            yolo, do it
          </OkuAlertDialogAction>
          <OkuAlertDialogCancel class="alert-dialog-cancel">
            maybe not
          </OkuAlertDialogCancel>
        </OkuAlertDialogContent>
      </OkuAlertDialogPortal>
    </OkuAlertDialog>
  </div>
</template>
