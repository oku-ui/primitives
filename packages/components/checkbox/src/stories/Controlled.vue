<script setup lang="ts">
import { ref } from 'vue'
import { OkuCheckbox, OkuCheckboxIndicator } from '@oku-ui/checkbox'
import Label from './Label.vue'

const checked = ref<boolean | 'indeterminate'>(true)
</script>

<template>
  <p>This checkbox is placed adjacent to its label. The state is controlled.</p>
  <Label for="randBox">Label</Label>{{ ' ' }}
  <OkuCheckbox
    id="randBox"
    class="checkbox"
    :checked="checked"
    @checked-change="checked = $event"
  >
    <OkuCheckboxIndicator class="checkbox-indicator" />
  </OkuCheckbox>
</template>
