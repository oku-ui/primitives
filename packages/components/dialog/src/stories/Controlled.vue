<script setup lang="ts">
import { OkuDialog, OkuDialogClose, OkuDialogContent, OkuDialogDescription, OkuDialogOverlay, OkuDialogPortal, OkuDialogTitle, OkuDialogTrigger } from '@oku-ui/dialog'
import { ref } from 'vue'

const open = ref(false)
</script>

<template>
  <OkuDialog v-model="open">
    <OkuDialogTrigger class="triggerClass">
      {{ open ? 'close' : 'open' }}
    </OkuDialogTrigger>
    <OkuDialogPortal>
      <OkuDialogOverlay class="overlayClass" />
      <OkuDialogContent class="contentDefaultClass">
        <OkuDialogTitle>
          Title
        </OkuDialogTitle>
        <OkuDialogDescription>
          Description
        </OkuDialogDescription>
        <OkuDialogClose>
          close
        </OkuDialogClose>
      </OkuDialogContent>
    </OkuDialogPortal>
  </OkuDialog>
</template>
