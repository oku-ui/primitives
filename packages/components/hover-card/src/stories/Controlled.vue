<script setup lang="ts">
import {
  OkuHoverCard,
  OkuHoverCardArrow,
  OkuHoverCardContent,
  OkuHoverCardPortal,
  OkuHoverCardTrigger,
} from '@oku-ui/hover-card'
import { ref } from 'vue'
import CardContentPlaceholder from './CardContentPlaceholder.vue'

const open = ref(false)
</script>

<template>
  <button @click="open = !open">
    Toggle
  </button>

  <div style="padding: 50px; display: flex; justify-content: center">
    <OkuHoverCard :open="open" @open-change="open = !open">
      <OkuHoverCardTrigger href="/" class="hover_card_triggerClass">
        trigger
      </OkuHoverCardTrigger>
      <OkuHoverCardPortal>
        <OkuHoverCardContent class="contentClass">
          <OkuHoverCardArrow class="hover_card_arrowClass" :width="20" :height="10" />
          <CardContentPlaceholder />
        </OkuHoverCardContent>
      </OkuHoverCardPortal>
    </OkuHoverCard>
  </div>
</template>
