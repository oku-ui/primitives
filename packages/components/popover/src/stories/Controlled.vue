<script setup lang="ts">
import {
  OkuPopover,
  OkuPopoverArrow,
  OkuPopoverClose,
  OkuPopoverContent,
  OkuPopoverPortal,
  OkuPopoverTrigger,
} from '@oku-ui/popover'
import { ref } from 'vue'

const open = ref(false)

function toggle(event: boolean) {
  open.value = event
}
</script>

<template>
  <div
    :style="{
      display: 'flex',
      alignItems: 'center',
      justifyContent: 'center',
      height: '50vh',
    }"
  >
    <OkuPopover :open="open" @open-change="toggle">
      <OkuPopoverTrigger class="triggerClass">
        {{ open ? 'close' : 'open' }}
      </OkuPopoverTrigger>
      <OkuPopoverPortal>
        <OkuPopoverContent class="contentClass">
          <OkuPopoverClose class="closeClass">
            close
          </OkuPopoverClose>
          <OkuPopoverArrow class="arrowClass" :width="20" :height="10" />
        </OkuPopoverContent>
      </OkuPopoverPortal>
    </OkuPopover>
  </div>
</template>
