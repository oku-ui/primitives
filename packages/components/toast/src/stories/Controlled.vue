<script setup lang="ts">
import {
  OkuToast, OkuToastAction,
  OkuToastDescription, OkuToastProvider, OkuToastViewport,
} from '@oku-ui/toast'
import { ref, watchEffect } from 'vue'
import { isClient } from '@oku-ui/use-composable'
import ToastUpgradeAvailable from './ToastUpgradeAvailable.vue'
import ToastSubscribeSuccess from './ToastSubscribeSuccess.vue'

const isSubscribed = ref(false)
const savedCount = ref(0)
const errorCount = ref(0)

const hasUpgrade = ref(false)

watchEffect((onInvalidate) => {
  if (!isClient)
    return
  if (!hasUpgrade.value) {
    const timer = window.setTimeout(() => hasUpgrade.value = true, 10000)

    onInvalidate(() => window.clearTimeout(timer))
  }
})
</script>

<template>
  <OkuToastProvider>
    <button type="button" class="px-4 py-2 bg-gray-200 hover:bg-gray-300 text-gray-800 text-sm font-medium rounded-md" @click="isSubscribed = true">
      subscribe
    </button>
    <button type="button" class="px-4 py-2 bg-gray-200 hover:bg-gray-300 text-gray-800 text-sm font-medium rounded-md" @click="errorCount++">
      error
    </button>
    <button type="button" class="px-4 py-2 bg-gray-200 hover:bg-gray-300 text-gray-800 text-sm font-medium rounded-md" @click="savedCount++">
      save
    </button>
    <ToastUpgradeAvailable v-model="hasUpgrade" />
    <ToastSubscribeSuccess v-model="isSubscribed" />

    <OkuToast v-for="(_, index) in errorCount" :key="index" class="toast error-toast">
      <OkuToastDescription>There was an error</OkuToastDescription>
      <OkuToastAction
        class="button" alt-text="Resubmit the form to try again."
        @click="console.log('try again')"
      >
        Try again
      </OkuToastAction>
    </OkuToast>

    <OkuToast v-for="(_, index) in savedCount" :key="index" class="toast">
      <OkuToastDescription>Successfully saved</OkuToastDescription>
    </OkuToast>

    <OkuToastViewport class="viewport" />
  </OkuToastProvider>
</template>
