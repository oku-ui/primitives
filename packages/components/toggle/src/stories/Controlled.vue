<script setup lang="ts">
import { ref } from 'vue'
import { OkuToggle } from '@oku-ui/toggle'

const pressed = ref(true)
</script>

<template>
  <OkuToggle class="toggle" :pressed="pressed" @pressed-change="pressed = $event">
    {{ pressed ? 'On' : 'Off' }}
  </OkuToggle>
</template>
