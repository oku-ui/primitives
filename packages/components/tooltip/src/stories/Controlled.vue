<script setup lang="ts">
import {
  OkuTooltip,
  OkuTooltipArrow,
  OkuTooltipContent,
  OkuTooltipPortal,
  OkuTooltipProvider,
  OkuTooltipTrigger,
} from '@oku-ui/tooltip'
import { ref } from 'vue'

const open = ref(true)
</script>

<template>
  <OkuTooltipProvider>
    <OkuTooltip v-model="open">
      <OkuTooltipTrigger class="triggerClass">
        I'm controlled, look I'm {{ open ? 'open' : 'closed' }}
      </OkuTooltipTrigger>
      <OkuTooltipPortal>
        <OkuTooltipContent class="contentClass" :side-offset="5">
          Nicely done!
          <OkuTooltipArrow class="arrowClass" :offset="10" />
        </OkuTooltipContent>
      </OkuTooltipPortal>
    </OkuTooltip>
  </OkuTooltipProvider>
</template>
