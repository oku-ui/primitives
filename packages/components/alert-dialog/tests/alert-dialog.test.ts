import { afterEach, beforeEach, describe, expect, it, vitest } from 'vitest'
import type { VueWrapper } from '@vue/test-utils'
import { enableAutoUnmount, mount } from '@vue/test-utils'
import { axe } from 'vitest-axe'
import { nextTick } from 'vue'
import {
  OkuAlertDialog,
  OkuAlertDialogAction,
  OkuAlertDialogCancel,
  OkuAlertDialogContent,
  OkuAlertDialogDescription,
  OkuAlertDialogDescriptionWarning,
  OkuAlertDialogOverlay,
  OkuAlertDialogPortal,
  OkuAlertDialogTitle,
  OkuAlertDialogTrigger,

} from '../src'

import StyledVue from '../src/stories/Styled.vue'
import Controlled from '../src/stories/Controlled.vue'

enableAutoUnmount(afterEach)

describe('OkuAlertDialog', () => {
  it('OkuAlertDialog renders correctly', () => {
    const wrapper = mount(OkuAlertDialog)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('OkuAlertDialogAction renders correctly', () => {
    const spy = vitest.spyOn(global.console, 'warn').mockImplementation(() => { })
    const wrapper = () => mount(OkuAlertDialogAction)
    expect(() => wrapper()).toThrowErrorMatchingSnapshot()
    expect(spy).toHaveBeenCalled()

    expect(spy.mock.calls[0][0]).toContain('[Vue warn]: injection "Symbol(OkuDialog)" not found.')
  })

  it('OkuAlertDialogCancel renders correctly', () => {
    const spy = vitest.spyOn(global.console, 'warn').mockImplementation(() => { })
    const wrapper = () => mount(OkuAlertDialogCancel)
    expect(() => wrapper()).toThrowErrorMatchingSnapshot()
    expect(spy).toHaveBeenCalled()

    expect(spy.mock.calls[0][0]).toContain('[Vue warn]: injection "Symbol(OkuAlertDialog)" not found.')
  })

  it('OkuAlertDialogContent renders correctly', () => {
    const spy = vitest.spyOn(global.console, 'warn').mockImplementation(() => { })
    const wrapper = () => mount(OkuAlertDialogContent)
    expect(() => wrapper()).toThrowErrorMatchingSnapshot()
    expect(spy).toHaveBeenCalled()

    expect(spy.mock.calls[0][0]).toContain('[Vue warn]: injection "Symbol(OkuDialog)" not found.')
  })

  it('OkuAlertDialogDescription renders correctly', () => {
    const spy = vitest.spyOn(global.console, 'warn').mockImplementation(() => { })
    const wrapper = () => mount(OkuAlertDialogDescription)
    expect(() => wrapper()).toThrowErrorMatchingSnapshot()
    expect(spy).toHaveBeenCalled()

    expect(spy.mock.calls[0][0]).toContain('[Vue warn]: injection "Symbol(OkuDialog)" not found.')
  })

  it('OkuAlertDialogDescriptionWarning renders correctly', () => {
    const wrapper = () => mount(OkuAlertDialogDescriptionWarning)
    expect(() => wrapper().html()).toMatchSnapshot()
  })

  it('OkuAlertDialogOverlay renders correctly', () => {
    const spy = vitest.spyOn(global.console, 'warn').mockImplementation(() => { })
    const wrapper = () => mount(OkuAlertDialogOverlay)
    expect(() => wrapper()).toThrowErrorMatchingSnapshot()
    expect(spy).toHaveBeenCalled()

    expect(spy.mock.calls[0][0]).toContain('[Vue warn]: injection "Symbol(OkuDialog)" not found.')
  })

  it('OkuAlertDialogPortal renders correctly', () => {
    const spy = vitest.spyOn(global.console, 'warn').mockImplementation(() => { })
    const wrapper = () => mount(OkuAlertDialogPortal)
    expect(() => wrapper()).toThrowErrorMatchingSnapshot()
    expect(spy).toHaveBeenCalled()

    expect(spy.mock.calls[0][0]).toContain('[Vue warn]: injection "Symbol(OkuDialog)" not found.')
  })

  it('OkuAlertDialogTitle renders correctly', () => {
    const spy = vitest.spyOn(global.console, 'warn').mockImplementation(() => { })
    const wrapper = () => mount(OkuAlertDialogTitle)
    expect(() => wrapper()).toThrowErrorMatchingSnapshot()
    expect(spy).toHaveBeenCalled()

    expect(spy.mock.calls[0][0]).toContain('[Vue warn]: injection "Symbol(OkuDialog)" not found.')
  })

  it('OkuAlertDialogTrigger renders correctly', () => {
    const spy = vitest.spyOn(global.console, 'warn').mockImplementation(() => { })
    const wrapper = () => mount(OkuAlertDialogTrigger)
    expect(() => wrapper()).toThrowErrorMatchingSnapshot()
    expect(spy).toHaveBeenCalled()

    expect(spy.mock.calls[0][0]).toContain('[Vue warn]: injection "Symbol(OkuDialog)" not found.')
  })

  describe('StyledVue', () => {
    let wrapper: VueWrapper<InstanceType<typeof StyledVue>>

    beforeEach(async () => {
      wrapper = mount(StyledVue, {
        attachTo: document.body,
      })

      await wrapper.find('[class="alert-dialog-trigger"]').trigger('click')
      await nextTick()
    })

    /**
    * @vitest-environment jsdom
    */
    it('axe accessibility tests', async () => {
      expect(await axe(wrapper.element)).toHaveNoViolations()
    })

    it('renders correctly', () => {
      expect(wrapper.html()).toMatchSnapshot()
    })

    it('should be able to open', async () => {
      expect(document.body).toMatchSnapshot()
    })

    it('should be able to close', async () => {
      await wrapper.find('[class="alert-dialog-trigger"]').trigger('click')
      await nextTick()
      expect(document.body).toMatchSnapshot()
    })
  })

  describe('Controlled', () => {
    let wrapper: VueWrapper<InstanceType<typeof Controlled>>

    beforeEach(async () => {
      wrapper = mount(Controlled, {
        attachTo: document.body,
      })

      await wrapper.find('[class="alert-dialog-trigger"]').trigger('click')
      await nextTick()
    })

    /**
     * @vitest-environment jsdom
      */

    it('axe accessibility tests', async () => {
      expect(await axe(wrapper.element)).toHaveNoViolations()
    })

    it('renders correctly', () => {
      expect(wrapper.html()).toMatchSnapshot()
    })

    it('should be able to open', async () => {
      expect(document.body).toMatchSnapshot()
    })

    it('should be able to close', async () => {
      await wrapper.find('[class="alert-dialog-trigger"]').trigger('click')
      await nextTick()
      expect(document.body).toMatchSnapshot()
    })
  })
})
