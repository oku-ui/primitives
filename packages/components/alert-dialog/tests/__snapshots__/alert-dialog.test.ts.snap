// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`okuAlertDialog > controlled > renders correctly 1`] = `
"<div>
  <div><img src="https://images.unsplash.com/photo-1682686580224-cd46ea1a6950?ixlib=rb-4.0.3&amp;ixid=M3wxMjA3fDF8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D&amp;auto=format&amp;fit=crop&amp;w=500&amp;q=80" alt="a large white house with a red roof"></div><button type="button" aria-haspopup="dialog" aria-expanded="true" aria-controls="oku-15" data-state="open" class="alert-dialog-trigger">Buy this house</button>
  <!--teleport start-->
  <!--teleport end-->
  <!--teleport start-->
  <!--teleport end-->
</div>"
`;

exports[`okuAlertDialog > controlled > should be able to close 1`] = `
<body
  style=""
>
  <div
    aria-hidden="true"
    data-aria-hidden="true"
    data-v-app=""
  />
  <div
    aria-hidden="true"
    data-aria-hidden="true"
    data-v-app=""
  />
  <div
    aria-hidden="true"
    data-aria-hidden="true"
    data-v-app=""
  />
  <div
    aria-hidden="true"
    data-aria-hidden="true"
    data-v-app=""
  />
  <div
    aria-hidden="true"
    data-aria-hidden="true"
    data-v-app=""
  />
  <div
    aria-hidden="true"
    data-aria-hidden="true"
    data-v-app=""
  />
  <div
    aria-hidden="true"
    data-aria-hidden="true"
    data-v-app=""
  />
  <div
    aria-hidden="true"
    data-aria-hidden="true"
    data-v-app=""
  >
    <div>
      <div>
        <img
          alt="a large white house with a red roof"
          src="https://images.unsplash.com/photo-1682686580224-cd46ea1a6950?ixlib=rb-4.0.3&ixid=M3wxMjA3fDF8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D&auto=format&fit=crop&w=500&q=80"
        />
      </div>
      
      <button
        aria-controls="oku-21"
        aria-expanded="false"
        aria-haspopup="dialog"
        class="alert-dialog-trigger"
        data-state="closed"
        type="button"
      >
        Buy this house
      </button>
      
      
      <!---->
      <!---->
      
      
      
    </div>
  </div>
</body>
`;

exports[`okuAlertDialog > controlled > should be able to open 1`] = `
<body
  style="pointer-events: none;"
>
  <span
    data-oku-focus-guard=""
    style="outline: none; opacity: 0; position: fixed; pointer-events: none;"
    tabindex="0"
  />
  <div
    aria-hidden="true"
    data-aria-hidden="true"
    data-v-app=""
  />
  <div
    aria-hidden="true"
    data-aria-hidden="true"
    data-v-app=""
  />
  <div
    aria-hidden="true"
    data-aria-hidden="true"
    data-v-app=""
  />
  <div
    aria-hidden="true"
    data-aria-hidden="true"
    data-v-app=""
  />
  <div
    aria-hidden="true"
    data-aria-hidden="true"
    data-v-app=""
  />
  <div
    aria-hidden="true"
    data-aria-hidden="true"
    data-v-app=""
  />
  <div
    data-v-app=""
  >
    <div>
      <div>
        <img
          alt="a large white house with a red roof"
          src="https://images.unsplash.com/photo-1682686580224-cd46ea1a6950?ixlib=rb-4.0.3&ixid=M3wxMjA3fDF8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D&auto=format&fit=crop&w=500&q=80"
        />
      </div>
      
      <button
        aria-controls="oku-18"
        aria-expanded="true"
        aria-haspopup="dialog"
        class="alert-dialog-trigger"
        data-state="open"
        type="button"
      >
        Buy this house
      </button>
      
      
      
      <!--teleport start-->
      <!--teleport end-->
      
      
      <!--teleport start-->
      <!--teleport end-->
      
      
      
      
    </div>
  </div>
  <div
    class="alert-dialog-overlay"
    data-state="open"
    style="pointer-events: auto;"
  >
    <!---->
  </div>
  
  
  <div
    aria-describedby="oku-20"
    aria-labelledby="oku-19"
    class="alert-dialog-content"
    data-state="open"
    id="oku-18"
    role="alertdialog"
    style="overflow: hidden; pointer-events: auto;"
    tabindex="-1"
  >
    
    <h2
      class="alert-dialog-title"
      id="oku-19"
    >
       Are you sure? 
    </h2>
    <p
      class="alert-dialog-description"
      id="oku-20"
    >
       This will do a very dangerous thing. Thar be dragons! 
    </p>
    <button
      class="alert-dialog-action"
      type="button"
    >
       yolo, do it 
    </button>
    <button
      class="alert-dialog-cancel"
      type="button"
    >
       maybe not 
    </button>
    
    <!---->
  </div>
  
  <!---->
  <!---->
  
  
  
  <span
    data-oku-focus-guard=""
    style="outline: none; opacity: 0; position: fixed; pointer-events: none;"
    tabindex="0"
  />
</body>
`;

exports[`okuAlertDialog > okuAlertDialog renders correctly 1`] = `""`;

exports[`okuAlertDialog > okuAlertDialogAction renders correctly 1`] = `[Error: \`OkuDialogClose\` must be used within \`OkuDialog\`]`;

exports[`okuAlertDialog > okuAlertDialogCancel renders correctly 1`] = `[Error: \`OkuAlertDialogCancel\` must be used within \`OkuAlertDialog\`]`;

exports[`okuAlertDialog > okuAlertDialogContent renders correctly 1`] = `[Error: \`OkuDialogContent\` must be used within \`OkuDialog\`]`;

exports[`okuAlertDialog > okuAlertDialogDescription renders correctly 1`] = `[Error: \`OkuDialogDescription\` must be used within \`OkuDialog\`]`;

exports[`okuAlertDialog > okuAlertDialogDescriptionWarning renders correctly 1`] = `[Function]`;

exports[`okuAlertDialog > okuAlertDialogOverlay renders correctly 1`] = `[Error: \`OkuDialogOverlay\` must be used within \`OkuDialog\`]`;

exports[`okuAlertDialog > okuAlertDialogPortal renders correctly 1`] = `[Error: \`OkuDialogPortal\` must be used within \`OkuDialog\`]`;

exports[`okuAlertDialog > okuAlertDialogTitle renders correctly 1`] = `[Error: \`OkuDialogTitle\` must be used within \`OkuDialog\`]`;

exports[`okuAlertDialog > okuAlertDialogTrigger renders correctly 1`] = `[Error: \`OkuDialogTrigger\` must be used within \`OkuDialog\`]`;

exports[`okuAlertDialog > styledVue > renders correctly 1`] = `
"<div><button type="button" aria-haspopup="dialog" aria-expanded="true" aria-controls="oku-3" data-state="open" class="alert-dialog-trigger"> delete everything </button>
  <!--teleport start-->
  <!--teleport end-->
  <!--teleport start-->
  <!--teleport end-->
</div>"
`;

exports[`okuAlertDialog > styledVue > should be able to close 1`] = `
<body
  style=""
>
  <div
    aria-hidden="true"
    data-aria-hidden="true"
    data-v-app=""
  />
  <div
    aria-hidden="true"
    data-aria-hidden="true"
    data-v-app=""
  />
  <div
    aria-hidden="true"
    data-aria-hidden="true"
    data-v-app=""
  />
  <div
    aria-hidden="true"
    data-aria-hidden="true"
    data-v-app=""
  >
    <div>
      
      <button
        aria-controls="oku-9"
        aria-expanded="false"
        aria-haspopup="dialog"
        class="alert-dialog-trigger"
        data-state="closed"
        type="button"
      >
         delete everything 
      </button>
      
      
      <!---->
      <!---->
      
      
      
    </div>
  </div>
</body>
`;

exports[`okuAlertDialog > styledVue > should be able to open 1`] = `
<body
  style="pointer-events: none;"
>
  <span
    data-oku-focus-guard=""
    style="outline: none; opacity: 0; position: fixed; pointer-events: none;"
    tabindex="0"
  />
  <div
    aria-hidden="true"
    data-aria-hidden="true"
    data-v-app=""
  />
  <div
    aria-hidden="true"
    data-aria-hidden="true"
    data-v-app=""
  />
  <div
    data-v-app=""
  >
    <div>
      
      <button
        aria-controls="oku-6"
        aria-expanded="true"
        aria-haspopup="dialog"
        class="alert-dialog-trigger"
        data-state="open"
        type="button"
      >
         delete everything 
      </button>
      
      
      
      <!--teleport start-->
      <!--teleport end-->
      
      
      <!--teleport start-->
      <!--teleport end-->
      
      
      
      
    </div>
  </div>
  <div
    class="alert-dialog-overlay"
    data-state="open"
    style="pointer-events: auto;"
  >
    <!---->
  </div>
  
  
  <div
    aria-describedby="oku-8"
    aria-labelledby="oku-7"
    class="alert-dialog-content"
    data-state="open"
    id="oku-6"
    role="alertdialog"
    style="overflow: hidden; pointer-events: auto;"
    tabindex="-1"
  >
    
    <h2
      class="alert-dialog-title"
      id="oku-7"
    >
       Are you sure? 
    </h2>
    <p
      class="alert-dialog-description"
      id="oku-8"
    >
       This will do a very dangerous thing. Thar be dragons! 
    </p>
    <button
      class="alert-dialog-action"
      type="button"
    >
       yolo, do it 
    </button>
    <button
      class="alert-dialog-cancel"
      type="button"
    >
       maybe not 
    </button>
    
    <!---->
  </div>
  
  <!---->
  <!---->
  
  
  
  <span
    data-oku-focus-guard=""
    style="outline: none; opacity: 0; position: fixed; pointer-events: none;"
    tabindex="0"
  />
</body>
`;
